modules/openstack/volume.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2023 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	services "github.com/gophercloud/gophercloud/openstack/blockstorage/extensions/services"
+)
+
+// ServiceCheck - Check particular cinder service is enabled and running or not
+func (o *OpenStack) ServiceCheck(
```

Comment:
@stuggi @konan-abhi wondering if this function can be considered generic enough to be moved in `service.go` : we'
re creating `volume.go`, but the function accepts a `serviceName string` as an argument, so it would make sense to me having it as part of service.go.

Reply:
IMO services module is related to keystone which is used for service related CRUD operations, we are using cinder extension here, so I think having this separate module makes more sense. Lets see what @stuggi has to say here.

Reply:
oh I see your point, "service" is an extension of `blockstorage`, therefore it justifies having a `volume.go` here.
I'm ok w/ this implementation then, and we can add more related to "volumes" as long as we need it.